docs/manual/resource-leak-checker.tex

```diff
@@ -0,0 +1,343 @@
+create an obligation, the checker may issue false positive warnings at calls to
+that method, but it never causes
+missed warnings (i.e., the checker is sound).
+
+\<@CreatesMustCallFor> allows the Resource Leak Checker to verify uses of non-final fields
+that contain a resource, even if they are re-assigned. Consider
+the following example:
+
+\begin{verbatim}
+  @MustCall("close") // default qualifier for uses of SocketContainer
+  class SocketContainer {
+    private @Owning Socket sock;
+
+    public SocketContainer() { sock = ...; }
+
+    void close() { sock.close() };
+
+    @CreatesMustCallFor("this")
+    void reconnect() {
+      if (!sock.isClosed()) {
+        sock.close();
+      }
+      sock = ...;
+    }
+  }
+\end{verbatim}
+
+In the lifetime of a \<SocketContainer> object, \<sock>
+might be re-assigned arbitrarily many times: once at each
+call to \<reconnect()>. This code is safe, however: \<reconnect()>
+ensures that \<sock> is closed before re-assigning it.
+
+Support for \<@CreatesMustCallFor> can be disabled by passing the \<-AnoCreatesMustCallFor> command-line
+argument to the checker. In this mode, \<@CreatesMustCallFor> annotations on methods are ignored at call-sites,
+and re-assigning a non-final, owning field is always an error.
+
+Sections~\ref{resource-leak-createsmustcallfor-callsite}
+and~\ref{resource-leak-createsmustcallfor-declaration}
+explain how the Resource Leak Checker verifies uses and declarations of
+methods annotated with \<@CreatesMustCallFor>.
+
+
+\subsectionAndLabel{Requirements at a call site of a \<@CreatesMustCallFor> method}{resource-leak-createsmustcallfor-callsite}
+
+At a call site to a method annotated as
+\<@CreatesMustCallFor(>\emph{expr}\<)>, the Resource Leak Checker:
+\begin{enumerate}
+\item
+  Treats any existing \<@MustCall> obligations of \emph{expr} as \emph{satisfied},
+\item
+  Creates a fresh obligation to check, as if \emph{expr} was a newly-allocated local
+  variable.
+\item
+  Requires that the expression corresponding to \emph{expr} (that is, \emph{expr}
+  viewpoint-adapted to the method call site) is owned; that is, it is
+  annotated or defaulted as \<@Owning>.  Otherwise, the checker
+  will issue a \<reset.not.owning> error at the call-site. You can avoid this
+  error by extracting the relevant expression into a new local variable (because
+  locals are \<@Owning> by default).
+\end{enumerate}
+
+Treating the obligation before the call as satisfied is sound: the
+checker creates a new obligation for calls to \<@CreatesMustCallFor> methods,
+and the Must Call Checker (\chapterpageref{must-call-checker}) ensures the
+\<@MustCall> type for the target will have a \emph{superset} of any methods
+present before the call. Intuitively, calling an \<@CreatesMustCallFor> method
+``resets'' the obligations of the target, so whether they were satisfied before
+the call or not is irrelevant.
+
+If an \<@CreatesMustCallFor>
```

Comment:
The referents in this paragraph are hard to follow.  Give the 3 methods names for concreteness.